tests/integration/test_partitioning_key.py

```diff
@@ -203,10 +203,11 @@
             # """
         ),
         (
-            [PartitionField(source_id=8, field_id=1001, transform=IdentityTransform(), name="timestamp_field")],
+            [PartitionField(source_id=8, field_id=1001, transform=IdentityTransform(), name="timestamp#field")],
             [datetime(2023, 1, 1, 12, 0, 1, 999)],
-            Record(timestamp_field=1672574401000999),
-            "timestamp_field=2023-01-01T12%3A00%3A01.000999",
+            Record(**{"timestamp#field": 1672574401000999}),  # type: ignore
```

Comment:
Mypy complains here (https://github.com/python/mypy/issues/5382) but I think it's ok